tulip/transys/transys.py

```diff
@@ -786,3 +789,165 @@ def __init__(self):
             'type?label': '',
             'separator': r'\\n'}
         self.dot_node_shape = {'normal': 'rectangle'}
+
+
+def simu_abstract(ts,simu_type):
+    """Create a bi/dual-simulation abstraction for a Finite Transition System.
+    
+    @param ts: input finite transition system, the one you want to get 
+                    its bi/dual-simulation abstraction.
+    @type ts: L{FTS}
+    @param simu_type: string 'bi'/'dual', flag used to switch b.w. 
+                      bisimulation algorithm and dual-simulation algorithm.
+    @return: the bi/dual simulation, and the corresponding partition.
+    @rtype: L{FTS}, C{dict}
+    
+    References
+    ==========
+    1. Wagenmaker, A. J.; Ozay, N.
+    A Bisimulation-like Algorithm for Abstracting Control Systems. 
+    54th Annual Allerton Conference on CCC 2016
+    """
+    def pre(graph,list_n):
```

Comment:
I support moving the function `pre` outside of the function `simu_abstract`, and hiding it by renaming to `_pre`. The definition of `pre` is independent of the arguments of `simu_abstract`, so this change is possible.